doc/CMakeLists.txt

```diff
@@ -16,4 +16,12 @@ install(
 	DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/src
     DESTINATION ${HTML_INSTALL_DIR}
 )
-endif(HIGH_FIVE_DOCUMENTATION)
\ No newline at end of file
+
+find_package(Doxygen)
+if(Doxygen_FOUND)
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile @ONLY)
+    add_custom_target(doxygen ALL COMMAND Doxygen::doxygen ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile)
+else()
+    message(STATUS "Doxygen documentation will not be build because Doxygen has not been found.")
```

Comment:
```suggestion
    message(WARNING "Doxygen documentation will not be built because Doxygen has not been found.")
```

Reply:
Plus I am not a big fan of building the documentation in target `all` in case it fails.
I would either add a CMake option to enable/disable Doxygen build or remove it from `all` target (I prefer the second solution).

Reply:
To have a documentation that build is a good thing imho, I like that travis said no if you break it.

Reply:
> To have a documentation that build is a good thing imho, I like that travis said no if you break it.

I agree. I suggest to execute `make doxygen` in Travis

Reply:
For travis the patch is easy but let's wait that the windows travis changes are merged.

Reply:
The `make doc` should be done in a new patchset